compiler/src/dotty/tools/repl/ParseResult.scala

```diff
@@ -116,7 +116,7 @@ object ParseResult {
   }
 
   /** Extract a `ParseResult` from the string `sourceCode` */
-  def apply(sourceCode: String)(implicit state: State): ParseResult =
+  def apply(sourceCode: String, sourceFile: Option[SourceFile] = None)(implicit state: State): ParseResult =
```

Comment:
We should not pass both the `sourceCode` and the `sourceFile` as the source file should contain the source code. 

You probably need:

```scala
def apply(sourceCode: String)(implicit state: State): ParseResult = 
 apply(SourceFile.virtual(str.REPL_SESSION_LINE + (state.objectIndex + 1), sourceCode))

def apply(sourceFile: SourceFile)(implicit state: State): ParseResult = {
  ...
}
```